components/object_store/src/obkv/util.rs

```diff
@@ -0,0 +1,19 @@
+use table_kv::{KeyBoundary, ScanRequest};
+
+/// Generate ScanRequest with prefix
+pub fn scan_request_with_prefix(prefix: &str) -> ScanRequest {
+    let key_buffer = prefix.as_bytes();
+    let mut start_key = Vec::with_capacity(key_buffer.len() + 1);
+    let mut end_key = Vec::with_capacity(key_buffer.len() + 1);
+    start_key.extend(key_buffer);
+    start_key.push(0x00);
+    end_key.extend(key_buffer);
+    end_key.push(0xff);
```

Comment:
In OBKV, is `0xFF00` less than `0xFF` and `0xFFFF` greater than `0xFFFF00`?

By intuition, I guess `0xFF00` is greater than `0xFF`, so here we can build the scan range is [start, MAX], and check the prefix of the results fetched from OBKV then.

Reply:
You mean the scan range should be [prefix, prefxi+[usize.max.to_](usize::MAX.to_be_bytes())]？

Reply:
> an the scan range should be` [prefix, prefxi+usize::MAX.to_be_bytes())]`？

Yes, but the caller should check the prefix if so. Another way to build the correct `end_key` by inc 1 to the byte array, but the overflow must be taken considerations.

Reply:
I rewrite this method, pls review it. thx.